images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh

```diff
@@ -462,6 +462,15 @@ patchBuildStep "${buildStartTime}" "${buildStartTime}" "${currentStepEnd}" "${NA
 previousStepEnd=${currentStepEnd}
 set -x
 
+##############################################
+### CACHE IMAGE LIST GENERATION
+##############################################
+
+LAGOON_CACHE_BUILD_ARGS=()
+readarray LAGOON_CACHE_BUILD_ARGS < <(kubectl --insecure-skip-tls-verify -n ${NAMESPACE} get deployments -o yaml -l 'lagoon.sh/service' | yq e '.items[].spec.template.spec.containers[].image | capture("^(?P<image>.+\/.+\/.+\/(?P<name>.+)\@.*)$") | "LAGOON_CACHE_" + .name + "=\"" + .image + "\""' -)
```

Comment:
we're getting this erroring on our version of yq
```
ci-nginx-control-k8s-nginx lagoon-build-72o75 jenkinslagoonpr29197-control-plane lagoon-build 2021-11-23T23:43:25.995103783Z Error: unknown command "e" for "yq"
```

exit status only appeared in yq 4 -we'll need to upgrade the version we use here (our version = 3.3.2, current release [4.15.1](https://github.com/mikefarah/yq/releases)

Reply:
@tobybellwood should we upgrade in this PR (can we?)

Reply:
I can probably get it to work without it as well

Reply:
maybe for now, working without yq4, as it's a bit of an upgrade.

Or do a second `yq4` binary, and we can switch the rest of the yq logic across later?